### PR TITLE
:technologist: Migrate follow management endpoints

### LINKED ISSUE #2855: API conventions for follower management endpoints
### Describe the bug

The endpoints for managing followers, followings, ... do not match the chosen naming conventions.

Therefore the following endpoints should be moved:

- [x] `GET /settings/followers` -> `GET /user/self/followers`
- [x] `DELETE /user/removeFollower` with userId in body -> `DELETE /user/self/followers/:id` without userId in body
- [x] `GET /settings/follow-requests` -> `GET /user/self/follow-requests`
- [x] `PUT /user/acceptFollowRequest` with userId in body -> `PUT /user/self/follow-requests/:userId` without userId in body
- [x] `DELETE /user/rejectFollowRequest` with userId in body -> `DELETE /user/self/follow-requests/:userId` without userId in body
- [x] `GET /settings/followings` -> `GET /user/self/followings`

~Not a bug but also useful to have and could be created when these endpoints are touched:~
~- `DELETE /settings/followings/:userId`~

Nevermind I could simply use the unfollow method for that lol

### Steps to reproduce

_No response_

### Browser console logs

_No response_

### Browser

_No response_

### PR DESCRIPTION
**Pull request by jheubuch**
closes #2855

### CODE REVIEW

**Comment by jheubuch** [comment]
Irgs I moved the `trusted-by` into the `user/self` group but this is breaking the tests? 🥲

File: storage/api-docs/api-docs.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
             "description": "Production Server"
         },
         {
-            "url": "http://localhost:8000/api/v1",
+            "url": "http://localhost:8000/api/v1/api/v1",
```
Comment (HerrLevin) [inline storage/api-docs/api-docs.json]: Your .env seems to be the old version. Please see 9025efc346af73ae6a94ae2c99f7b994930440cd

File: storage/api-docs/api-docs.json [hunk 2]
```diff
@@ -105,7 +105,7 @@
                     "Auth"
                 ],
                 "summary": "Refresh Bearer Token",
-                "description": "This request issues a new Bearer-Token with a new expiration date while also revoking the old\n     *      token.",
+                "description": "This request issues a new Bearer-Token with a new expiration date while also revoking the old\r\n     *      token.",
```
Comment (HerrLevin) [inline storage/api-docs/api-docs.json:108]: Spotted the windows user 👀

**Review by HerrLevin — APPROVED** [review]

**Comment by marhei** [comment]
One (late) addition: Since the user endpoints are now streamlined, they should imo be in the same category in the API docs: `/user/self/followers`, `/user/self/follow-requests` and `/user/self/followings` are still part of `Settings` and not `User/Follow`, which even describes itself as the category to manage your followers...

**Pull request merged**